### PR TITLE
Initial NPI-19 pressure sensor documentation

### PR DESCRIPTION
**Pull request by bakerkj**
## Description:

**Related issue (if applicable): none

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#7181

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by nagyrobi** [comment]
As this is a feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase.

Thanks

**Comment by bakerkj** [comment]
> As this is a feature matched with a PR in `esphome/esphome`, please target your PR to the `next` branch and rebase.

Sorry. I missed that. Now it is targeted against next.

Thanks!

File: components/sensor/npi19.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Converting units
+-----------------
+
+The NPI-19 pressure sensor is not calibrated to units, you have to convert the measurement to units yourself.
+
+Estimated
+*********
+
+On page 1 of the `product application note <https://www.amphenol-sensors.com/hubfs/I2C%20NPI-19%20product%20application%20Note.pdf>`__
+the value ``1638`` maps to approximately ``10%`` of the maximum value of the sesnor (e.g. ``0.5`` psi for a ``5`` psi sensor);
```
Comment (jesserockz) [inline components/sensor/npi19.rst]:
```suggestion
the value ``1638`` maps to approximately ``10%`` of the maximum value of the sensor (e.g. ``0.5`` psi for a ``5`` psi sensor);
```

File: components/sensor/npi19.rst [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        source_id: raw_pressure
+        name: Pressure
+        state_class: measurement
+        device_class: pressure
+        unit_of_measurement: psi
+
+        filters:
+          - calibrate_linear:
+            - 1638 -> 0.5
+            - 14746 -> 4.5
```
Comment (jesserockz) [inline components/sensor/npi19.rst]: Dont need extra info in the example. If people read the page properly they will see the linear calibration suggested below. If you want, you can include the `copy` sensor in the calibration section

```suggestion
    # Example configuration entry
    sensor:
      - platform: npi19
        raw_pressure:
          name: "Raw Pressure"
        temperature:
          name: Temperature
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**